packages/puppeteer-core/src/common/bidi/Page.ts

```diff
@@ -241,8 +237,11 @@ export class BidiPage extends Page {
 
   #onFrameLoaded(info: Bidi.BrowsingContext.NavigationInfo): void {
     const frame = this.frame(info.context);
-    if (frame && this.mainFrame() === frame) {
-      this.emit(PageEmittedEvents.Load);
+    if (frame) {
+      if (this.mainFrame() === frame) {
+        this.emit(PageEmittedEvents.Load);
+      }
+      this.emit(PageEmittedEvents.FrameNavigated, frame);
```

Comment:
This is not correct. Or at least there is better way of doing it then what we currently have and that is DomContentLoaded event